sdk/cosmos/azure_data_cosmos/src/query/executor.rs

```diff
@@ -0,0 +1,202 @@
+use azure_core::http::{headers::Headers, Context, Method, Request, Response};
+use serde::de::DeserializeOwned;
+
+use crate::{
+    constants,
+    pipeline::{self, CosmosPipeline},
+    query::{OwnedQueryPipeline, QueryEngineRef, QueryResult},
+    resource_context::{ResourceLink, ResourceType},
+    FeedPage, FeedPager, Query, QueryOptions,
+};
+
+pub struct QueryExecutor<T: DeserializeOwned> {
+    http_pipeline: CosmosPipeline,
+    container_link: ResourceLink,
+    items_link: ResourceLink,
+    context: Context<'static>,
+    query_engine: QueryEngineRef,
+    base_request: Request,
+    query: Query,
+    pipeline: Option<OwnedQueryPipeline>,
+
+    // Why is our phantom type a function? Because that represents how we _use_ the type T.
+    // Normally, PhantomData<T> is only Send/Sync if T is, because PhantomData is indicating that while we don't _name_ T in a field, we should act as though we have a field of type T.
+    // However, we don't store any T values in this, we only RETURN them.
+    // That means we use a function pointer to indicate that we don't actually operate on T directly, we just return it.
+    // Because of this, PhantomData<fn() -> T> is Send/Sync even if T isn't (see https://doc.rust-lang.org/stable/nomicon/phantom-data.html#table-of-phantomdata-patterns)
+    phantom: std::marker::PhantomData<fn() -> T>,
```

Comment:
No lint on this?
```suggestion
    _phantom: std::marker::PhantomData<fn() -> T>,
```

Reply:
Nope! `PhantomData` is specifically handled in the dead-code analysis pass (https://github.com/rust-lang/rust/blob/dcecb99176edf2eec51613730937d21cdd5c8f6e/compiler/rustc_passes/src/dead.rs#L944). A field of type `PhantomData` is always considered live, even if it's never read.